tcms/testplans/tests/test_new_plan.py

```diff
@@ -96,16 +126,87 @@ def _test_plan_create_new(self, is_active):
+            response,
+            _(
+                "Select a valid choice. That choice is not one of the available choices."
+            ),
+        )
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="email_settings-0-auto_to_plan_author" '
+            'type="checkbox"',
+            html=False,
+        )
+
 
-        response = self.client.get(self.location, follow=True)
+class TestNewTestPlanViewPermissions(PermissionsTestCase):
+    add_testplan_permission = "testplans.add_testplan"
+    permission_label = add_testplan_permission
+    http_method_names = ["get", "post"]
+    url = reverse("plans-new")
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.post_data = {"description": "test"}
+        super().setUpTestData()
+
+        cls.plan_name = 'plan_name'
+        cls.product = ProductFactory()
+        cls.product_version = VersionFactory(product=cls.product)
+        cls.plan_type = PlanTypeFactory()
+        cls.test_plan = TestPlanFactory()
+
+        user_should_have_perm(cls.tester, perm="testplans.view_testplan")
+
+        cls.post_data.update(
+            {
+                "author": cls.tester.pk,
+                "product": cls.product.pk,
+                "product_version": cls.product_version.pk,
+                "type": cls.plan_type.pk,
+                "name": cls.plan_name,
+                "email_settings-0-auto_to_plan_author": "on",
+                "email_settings-0-auto_to_case_owner": "on",
+                "email_settings-0-auto_to_case_default_tester": "on",
+                "email_settings-0-notify_on_case_update": "on",
+                "email_settings-0-notify_on_plan_update": "on",
+                "email_settings-0-id": cls.test_plan.emailing.pk,
+                "email_settings-TOTAL_FORMS": "1",
+                "email_settings-INITIAL_FORMS": "1",
+                "email_settings-MIN_NUM_FORMS": "0",
+                "email_settings-MAX_NUM_FORMS": "1",
+                "is_active": True,
+            }
+        )
 
-        self.assertRedirects(response, reverse('tcms-login') + '?next=' + self.location)
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
 
-    def test_post_with_no_perm_redirects_to_login(self):
-        remove_perm_from_user(self.user, self.add_testplan_permission)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _('Create new TestPlan'))
+
+    def verify_post_with_permission(self):
+        response = self.client.post(self.url, self.post_data, follow=True)
+        test_plan = TestPlan.objects.get(
+            name=self.post_data["name"],
+            product=self.post_data["product"],
+        )
+        redirect_url = reverse('test_plan_url', args=[test_plan.pk, slugify(test_plan.name)])
 
-        response = self.client.post(self.location, self.request, follow=True)
+        self.assertRedirects(response, redirect_url)
```

Comment:
There is also a short URL (/plan/123/) which itself redirects to the one with the slugified name. IDK if assertRedirects will work if you try the short URL but that may save you from using slugify in the tests.

Reply:
I did try using that but it doesn't work with the short URL.